pkg/reconciliation/reconcile_racks.go

```diff
@@ -780,9 +780,9 @@ func (rc *ReconciliationContext) CheckRackScale() result.ReconcileResult {
 				updated = rc.setCondition(
 					api.NewDatacenterCondition(
 						api.DatacenterResuming, corev1.ConditionTrue)) || updated
-			} else {
+			} else if dc.GetConditionStatus(api.DatacenterReady) == corev1.ConditionTrue {
 				// We weren't resuming from a stopped state, so we must be growing the
-				// size of the rack
+				// size of the rack and this isn't the initialization stage
```

Comment:
Currently (in master) the scaling up condition does get set to true initially before the datacenter is ready.  Are there any scenarios in which the Ready condition could be false and scaling up would happen? If so would it be better/safer to handle this in the `CheckClearActionConditions` function. The function could be updated to only call `cleanupAfterScaling` after the Initialized condition is true.

Reply:
No, there's no such thing. This is the only place where ScalingUp is set (and I don't think it should be set when the cluster is starting).

Reply:
My main point is that ScalingUp is set before the datacenter becomes ready, or more precisely, before the Initialized condition is set. What is the rationale for changing that existing behavior?

Reply:
Because creating a new cluster is not scaling up existing one. I don't feel like these two should share the same status - especially if we have a different process for both cases.